cloud/pkg/apis/edgeapplication/v1alpha1/type.go

```diff
@@ -0,0 +1,221 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=`.status.conditions[?(@.type=="Applied")].status`,name="Applied",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+
+// EdgeApplication defines a list of resources to be deployed on the node groups.
+type EdgeApplication struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the desired behavior of EdgeApplication.
+	Spec EdgeAppSpec `json:"spec"`
+
+	// Status represents the status of PropagationStatus.
+	// +optional
+	Status EdgeAppStatus `json:"status,omitempty"`
+}
+
+// EdgeAppSpec defines the desired state of EdgeApplication.
+type EdgeAppSpec struct {
+	// ResourceTemplate represents the manifest workload to be deployed on managed node groups.
+	ResourceTemplate ResourceTemplate `json:"resourceTemplate,omitempty"`
+	// WorkloadScopes represents the scope of workload to be deployed.
+	WorkloadScopes WorkloadScope `json:"workloadScopes"`
+}
+
+// WorkloadScope represents the scope of workload to be deployed.
+type WorkloadScope struct {
+	// TargetNodeGroups represents the target node groups of workload to be deployed.
+	// +optional
+	TargetNodeGroups []TargetNodeGroups `json:"targetNodeGroups,omitempty"`
+	// TargetNodes represents the target nodes of workload to be deployed.
+	// +optional
+	TargetNodes []TargetNodes `json:"targetNodes,omitempty"`
+}
+
+// TargetNodeGroups represents the target node groups of workload to be deployed.
+type TargetNodeGroups struct {
+	// Name represents the name of target node group
+	Name string `json:"name"`
+	// Overriders offers various alternatives to represent the override rules.
+	Overriders Overriders `json:"overriders,omitempty"`
+}
+
+// TargetNodes represents the target nodes of workload to be deployed.
+type TargetNodes struct {
+	// Label of target nodes
+	Label labels.Selector `json:"label"`
+	// Overriders offers various alternatives to represent the override rules.
+	Overriders Overriders `json:"overriders"`
+}
+
+// ResourceTemplate represents the manifest workload to be deployed on managed node groups.
+type ResourceTemplate struct {
+	// Manifests represents a list of Kubernetes resources to be deployed on the managed node groups.
+	// +optional
+	Manifests []Manifest `json:"manifests,omitempty"`
+}
+
+// Overriders offers various alternatives to represent the override rules.
+//
+// If more than one alternatives exist, they will be applied with following order:
+// - ImageOverrider
+// - Plaintext
```

Comment:
It seems that no Plaintext in the `Overriders` struct. So, I think there's no need to give the description of the priority.

And should we support plaintext in the later release?

Reply:
yes, there is no need to give the priority description here now, we can support plaintext if users have requirements in the future, and I think imageOverrider can be achieved first.